### ISSUE TITLE
Result of `Series.mean()` cannot be used as `float`

### ISSUE THREAD

**Issue by Azureblade3808** [pandas-dev/pandas-stubs#646]
**Describe the bug**
As the title.

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.
```python
import pandas as pd

series = pd.Series([0, 1, 2])
mean: float = series.mean()  # error
```

1. Indicate which type checker you are using (`mypy` or  `pyright`).
`pyright`

1. Show the error message received from that type checker while checking your example.
```
"float64" is incompatible with "float"
```

**Please complete the following information:**
 - OS: [e.g. Windows, Linux, MacOS]
Windows

 - OS Version [e.g. 22]
10

 - python version
3.10.10

 - version of type checker
1.1.301 (bundled with pylance)

 - version of installed `pandas-stubs`
2.0.0.230412

**Additional context**
Per #552 and #622 , `Series.mean()` is changed into return `numpy.float64` instead of `float`.

Although `numpy.float64` is a subtype of `float` in runtime most of the time (on 64bit systems), it's not technically so, and there is no evidence for that in numpy stubs ("numpy/\_\_init\_\_.pyi"). Therefore, a `numpy.float64` value would not be allowed to be used as `float` by a type-checker.

Considering how most APIs here are declared to return `int`s or `float`s instead of `numpy.int64`s or `numpy.float64`s, I think that it would be fair to say that these returned values are more commonly used as if they are plain `int`s or `float`s. I'd like to see the change be reverted.

**Comment by Dr-Irv** [comment]
> Although `numpy.float64` is a subtype of `float` in runtime most of the time (on 64bit systems), it's not technically so, and there is no evidence for that in numpy stubs ("numpy/__init__.pyi"). Therefore, a `numpy.float64` value would not be allowed to be used as `float` by a type-checker.

Can you explain what you mean by "it's not technically so" ?

> 
> Considering how most APIs here are declared to return `int`s or `float`s instead of `numpy.int64`s or `numpy.float64`s, I think that it would be fair to say that these returned values are more commonly used as if they are plain `int`s or `float`s. I'd like to see the change be reverted.

Well, for `Series.mean()`, we document it as returning a scalar.

Having said that, if using the `pyarrow` backend, the result is a `float`:
```python
>>> import pandas as pd
>>> s=pd.Series([1,2.2,3])
>>> pa = pd.to_numeric(s, dtype_backend="pyarrow")
>>> pa
0    1.0
1    2.2
2    3.0
dtype: double[pyarrow]
>>> ma = pa.mean()
>>> type(ma)
<class 'float'>
```

So I agree with you that we can revert this change.  PR welcome.

@ramvikrams can you do this?

**Comment by gandhis1** [comment]
Is the right thing to do to sub-class frame/series for the pyarrow backend, and then overload the constructor and read functions to create one?

**Comment by ramvikrams** [comment]
> So I agree with you that we can revert this change. PR welcome.
> 
> @ramvikrams can you do this?

Surely

**Comment by Dr-Irv** [comment]
closed via #647
